src/main/java/org/redcrosswarriors/controller/RequestBloodController.java

```diff
@@ -37,14 +37,23 @@
 
     @PostMapping("/requestBlood")
     @Secured("ROLE_USER")
-    public ResponseEntity<Object> requestBlood(@RequestBody RequestBloodInput input, Principal principal)
-    {
+    public ResponseEntity<Object> requestBlood(@RequestBody RequestBloodInput input, Principal principal) {
```

Comment:
If you want the validation that you added in the RequestInputDetails  to work here you would need to change RequestBloodInput to ReqeustInputDetaials and add @Valid

Reply:
However if you want to keep it the way it is that's fine too.